libp2p/muxers/mplex/mplex.nim

```diff
@@ -154,5 +153,10 @@ method newStream*(m: Mplex,
 
 method close*(m: Mplex) {.async, gcsafe.} =
   trace "closing mplex muxer"
+  if not m.connection.closed():
+    await m.connection.close()
+
   await allFutures(@[allFutures(toSeq(m.remote.values).mapIt(it.reset())),
```

Comment:
might help to merge this https://github.com/status-im/nim-libp2p/pull/125 but anyway can be the opposite way too

Reply:
This is probably more urgent since it fixes some pretty egregious mem leaks. I'd rather get this in first to see how it behaves in NBC.